interoperator/pkg/controller/sfplan/sfplan_controller.go

```diff
@@ -65,18 +74,39 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
+func restartOnWatchUpdate(mgr manager.Manager, initWatches chan bool) {
+	for {
+		select {
+		case <-initWatches:
```

Comment:
Shouldn't we select from `stopCh` as well to handle the graceful termination?

Reply:
Added